src/steps/email-links-validation.ts

```diff
@@ -84,32 +93,32 @@ export class EmailLinksValidationStep extends BaseStep implements StepInterface
       const plainUrls = Array.from(GetUrls(plain).values()).map((f) => { return { url: f, type: 'Plain' }; });
 
       const urls = new Set(htmlUrls.concat(plainUrls));
+      const sanitizedUrls = this.sanitizeUrl(Array.from(urls.values()));
 
       const response = await this.client.evaluateUrls(
-        this.sanitizeUrl(Array.from(urls.values())));
-
-      if (response.length > 0) {
-        const plain = response.filter(f => f.type === 'Plain');
-        const html = response.filter(f => f.type === 'HTML');
-        return this.fail('Broken links were found in the email. URLs include: %s%s%s%s%s', [
-          `${nl}`,
-          `${nl}Plain Text: ${nl}`,
-          plain.length > 0 ? plain.map(f => `${f.url} (${f.message})`.trim()).join(`${nl}`) : `No URLs found in Plain Text Body${nl}`,
-          `${nl}${nl}HTML: ${nl}`,
-          html.length > 0 ? html.map(f => `${f.url} (${f.message})`.trim()).join(`${nl}`) : `No URLS found in HTML Body${nl}`,
-        ]);
+        sanitizedUrls,
+      );
+
+      const brokenUrls = response.brokenUrls;
+      const allUrls = response.brokenUrls.concat(response.workingUrls);
+      const linkRecords = this.createLinkRecords(allUrls);
+
+      if (brokenUrls.length > 0) {
+        const plain = brokenUrls.filter(f => f.type === 'Plain');
+        const html = brokenUrls.filter(f => f.type === 'HTML');
+        return this.fail('Broken links were found in the email', [], [messageRecords, linkRecords]);
```

Comment:
- [ ] Recommend passing `linkRecords` first to better tie to the "Broken links were found in the email" message.  Otherwise the `messageRecords` table shows up first and it's a little confusing.